File: app/src/main/java/com/rhdevs/rhpatch/youtube/extension/youtube/patches/ForceOriginalAudioPatch.java
package com.rhdevs.rhpatch.youtube.extension.youtube.patches;

import com.rhdevs.rhpatch.youtube.extension.youtube.settings.Settings;

@SuppressWarnings("unused")
public class ForceOriginalAudioPatch {

    /**
     * Injection point.
     */
    public static void setEnabled() {
        com.rhdevs.rhpatch.youtube.extension.shared.patches.ForceOriginalAudioPatch.setEnabled(
                Settings.FORCE_ORIGINAL_AUDIO.get(),
                Settings.SPOOF_VIDEO_STREAMS_CLIENT_TYPE.get()
        );
    }
}

File: app/src/main/java/com/rhdevs/rhpatch/morphe/youtube/video/audio/ForceOriginalAudioPatch.kt
package com.rhdevs.rhpatch.morphe.youtube.video.audio

import app.morphe.extension.youtube.patches.ForceOriginalAudioPatch
import com.rhdevs.rhpatch.morphe.shared.misc.audio.tracks.forceOriginalAudioPatch
import com.rhdevs.rhpatch.morphe.youtube.misc.playservice.VersionCheck
import com.rhdevs.rhpatch.morphe.youtube.misc.playservice.is_21_26_or_greater
import com.rhdevs.rhpatch.morphe.youtube.misc.settings.PreferenceScreen
import com.rhdevs.rhpatch.morphe.youtube.shared.YouTubeActivityOnCreateFingerprint

val ForceOriginalAudio = forceOriginalAudioPatch(
    block =  {
        dependsOn(
            VersionCheck
        )
    },
    // Localized audio track flag was removed in 21.26+ but might be replaced with 45673827L
    fixUseLocalizedAudioTrackFlag = { !is_21_26_or_greater },
    mainActivityOnCreateFingerprint = YouTubeActivityOnCreateFingerprint,
    subclassExtensionSetEnabled = ForceOriginalAudioPatch::setEnabled,
    preferenceScreen = PreferenceScreen.VIDEO,
)
